app/javascript/hooks/request-query.ts

```diff
@@ -10,17 +9,17 @@ import { camelizeKeys } from 'humps'
 
 type Request = {
   endpoint: string
-  query?: string
+  query?: object
```

Comment:
Probably not accepting _any_ object, linter/ts should have complained about this.

```suggestion
  query?: Record<string, string>
```

Or since you use it as such:

```suggestion
  query?: Parameters<UrlParams>[0]
```

Reply:
Never new about `Parameters`! Awesome. I found that there is even a `ConstructorParameters` option.